### PR TITLE
Show vertical payment method list in vertical mode

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
Lots of TODOs but adds basic saved PM / new PM list to vertical mode main screen.

## Testing
See snapshot tests

## Changelog
Not user facing

### CODE REVIEW

**Comment by porter-stripe** [comment]
Looks like the margins on the row button got unintentionally changed, but did correctly catch that the label wasn't reading the appearance `componentText`!

**Comment by yuki-stripe** [comment]
> Looks like the margins on the row button got unintentionally changed, but did correctly catch that the label wasn't reading the appearance `componentText`!

Margins still don't quite match the Figma, will leave it to another PR that makes PaymentMethodRowButton use `RowButton` internally to fix.

File: StripePaymentSheet/StripePaymentSheet.xcodeproj/xcshareddata/xcschemes/StripePaymentSheet.xcscheme [hunk 1]
```diff
@@ -42,6 +42,11 @@
             value = "$(SRCROOT)/../Tests/ReferenceImages"
             isEnabled = "YES">
          </EnvironmentVariable>
+         <EnvironmentVariable
+            key = "recordMode"
+            value = "false"
+            isEnabled = "YES">
+         </EnvironmentVariable>
```
[the remark this answers is not in the document]
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet.xcodeproj/xcshareddata/xcschemes/StripePaymentSheet.xcscheme:49]: I believe it's another way to toggle record mode for snapshot tests but I'll remove, not sure it works.

[the remark this answers is not in the document]
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet.xcodeproj/xcshareddata/xcschemes/StripePaymentSheet.xcscheme:49]: Yeah it's actually "STP_RECORD_SNAPSHOTS", not "recordMode"

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Views/PaymentSheetUIKitAdditions.swift [hunk 2]
```diff
@@ -141,3 +141,28 @@ extension UIFont {
         return UIFont(descriptor: descriptor, size: pointSize)
     }
 }
+
+extension UILabel {
+    static func makeVerticalRowButtonLabel(text: String, appearance: PaymentSheet.Appearance) -> UILabel {
+        let label = UILabel()
+        label.font = appearance.scaledFont(for: appearance.font.base.regular, style: .subheadline, maximumPointSize: 25)
```
[the remark this answers is not in the document]
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Views/PaymentSheetUIKitAdditions.swift:148]: Ah yeah, should be medium. I thought it looked too heavy but that's b/c the current font size is 15 vs figma's 14.  I think we need to do some custom thing to get it to respect dynamic type and also be size 14 by default. https://jira.corp.stripe.com/browse/MOBILESDK-2079

**Review by porter-stripe — APPROVED** [review]

**Review by porter-stripe — APPROVED** [review]

**Pull request merged**